docs/general-concepts/forms-fields/standard-fields/modalselect.md

```diff
@@ -0,0 +1,221 @@
+- **hint** (optional) The HTML placeholder text 
+
+- **select** set to "true" if you want to be able to select an item 
+- **iconSelect** the [icon](../../icons.md) to display beside the "Select". Default is "icon-file"
+- **titleSelect** the text (translatable) to display at the top of the Select modal
+- **urlSelect** the URL for the iframe for the select modal window
+
+- **sql_title_table**, **sql_title_column**, **sql_title_key** you'll probably need to set these fields for your ModalSelect field to work properly.
+They're necessary if you're redisplaying a form after a submit. 
+This is because in the submit POST parameters there's only room for the modal select field id, but when it's redisplayed it shows the associated title also.
+So these fields are used to obtain the title from the id, by performing a SQL query on the database table.
+
+- **clear** set to "true" if you want to allower users to cancel the selection of an item 
+This causes the Clear button to be displayed when an item is selected.
+
+- **new** set to "true" if you want to be able to create a new item
+This causes the Create button to be displayed when no item is selected.
+- **titleNew** the text (translatable) to display at the top of the Create modal
+- **urlNew** the URL for the iframe for the create modal
+
+- **edit** set to "true" if you want to be able to edit a selected item
+This causes the Edit button to be displayed when an item is selected.
+- **titleEdit** the text (translatable) to display at the top of the Edit modal
+- **urlEdit** the URL for the iframe for the edit modal
+
+- **urlCheckin** the URL to use to check-in an item
+
+- **onchange** if you're using the ModalSelectField inside filter fields then you'll probably want
+to set this to "this.form.submit();".
+This means that once an item is selected the filter fields form is submitted to redisplay the
+records based on the updated filter criteria.
+
+Implemented by: libraries/src/Form/Field/ModalSelectField.php
+
+## Example ModalSelect field
+
+ModalSelect fields work best on the administrator back-end, because that's where the necessary Joomla code is located for working with them.
+
+However, you can use this [exampleform component](../../../building-extensions/components/example-form-component.md) and insert the following into the XML form.
+
+```xml
+<field
+  type="ModalSelect"
+  name="article"
+  label="Article"
+  select="true"
+  iconSelect="fa fa-bomb"
+  titleSelect="Selecting the Article"
+  urlSelect="index.php?option=com_content&amp;view=articles&amp;layout=modal&amp;tmpl=component"
+  clear="true"
+  sql_title_table="#__content"
+  sql_title_column="title"
+  sql_title_key="id"
+/>
+```
+
+By navigating to "/index.php/component/exampleform/" on your site front-end, this will display a ModalSelect field which will allow you to select an article.
+
+This works because in the site com_content DisplayController constructor it has:
```

Comment:
I not very understood this part. What you tries to explain? :wink: 
`ModalSelect` field is "global" field, provided by library, it will work for every part of the site which uses a Form, without extra work.

Reply:
Ah I get it, it about the popup content, not about the field.
Maybe better something like this (with better wording):

-------

By navigating to "/index.php/component/exampleform/" on your site front-end, this will display a ModalSelect field which will allow you to select an article.

After clicking the "select" button, the field will open dialog with iframe with content from `urlSelect`, whish is `index.php?option=com_content&view=articles&layout=modal&tmpl=component`, with list of the articles to select. In this example, on the front-end the com_content view will assign the `modal` layout from administrator side, with following code: ....
And on the administrator side com_content will use `modal` layout dircetly,

Reply:
Thanks for the suggestions, I've changed the text along the lines you suggested.

I thought it might be best to move the "How it Works" section up as well.

Reply:
Thanks!

> I thought it might be best to move the "How it Works" section up as well.

No need, the example is also part of "how it works", it is good here.